pkg/cmd/pr/checks/checks.go

```diff
@@ -83,163 +93,75 @@ func checksRun(opts *ChecksOptions) error {
 
 	isTerminal := opts.IO.IsStdoutTTY()
 
-	if opts.WebMode {
-		openURL := ghrepo.GenerateRepoURL(baseRepo, "pull/%d/checks", pr.Number)
-		if isTerminal {
-			fmt.Fprintf(opts.IO.ErrOut, "Opening %s in your browser.\n", utils.DisplayURL(openURL))
-		}
-		return opts.Browser.Browse(openURL)
-	}
-
-	if len(pr.StatusCheckRollup.Nodes) == 0 {
-		return fmt.Errorf("no commit found on the pull request")
+	openURL := ghrepo.GenerateRepoURL(baseRepo, "pull/%d/checks", pr.Number)
+	if isTerminal {
+		fmt.Fprintf(opts.IO.ErrOut, "Opening %s in your browser.\n", utils.DisplayURL(openURL))
 	}
+	return opts.Browser.Browse(openURL)
+}
 
-	rollup := pr.StatusCheckRollup.Nodes[0].Commit.StatusCheckRollup.Contexts.Nodes
-	if len(rollup) == 0 {
-		return fmt.Errorf("no checks reported on the '%s' branch", pr.BaseRefName)
+func checksRunWatchMode(opts *ChecksOptions) error {
+	duration, err := time.ParseDuration(fmt.Sprintf("%ds", opts.Interval))
+	if err != nil {
+		return fmt.Errorf("could not parse interval: %w", err)
 	}
 
-	passing := 0
-	failing := 0
-	skipping := 0
-	pending := 0
-
-	type output struct {
-		mark      string
-		bucket    string
-		name      string
-		elapsed   string
-		link      string
-		markColor func(string) string
+	checks, meta, err := collect(opts)
+	if err != nil {
+		return fmt.Errorf("cannot collect checks information: %w", err)
 	}
 
+	isTerminal := opts.IO.IsStdoutTTY()
 	cs := opts.IO.ColorScheme()
 
-	outputs := []output{}
-
-	for _, c := range pr.StatusCheckRollup.Nodes[0].Commit.StatusCheckRollup.Contexts.Nodes {
-		mark := "✓"
-		bucket := "pass"
-		state := c.State
-		markColor := cs.Green
-		if state == "" {
-			if c.Status == "COMPLETED" {
-				state = c.Conclusion
-			} else {
-				state = c.Status
-			}
-		}
-		switch state {
-		case "SUCCESS":
-			passing++
-		case "SKIPPED", "NEUTRAL":
-			mark = "-"
-			markColor = cs.Gray
-			skipping++
-			bucket = "skipping"
-		case "ERROR", "FAILURE", "CANCELLED", "TIMED_OUT", "ACTION_REQUIRED":
-			mark = "X"
-			markColor = cs.Red
-			failing++
-			bucket = "fail"
-		default: // "EXPECTED", "REQUESTED", "WAITING", "QUEUED", "PENDING", "IN_PROGRESS", "STALE"
-			mark = "*"
-			markColor = cs.Yellow
-			pending++
-			bucket = "pending"
-		}
+	err = printTable(isTerminal, cs, meta, checks, opts)
+	if err != nil {
+		return fmt.Errorf("cannot print table: %w", err)
+	}
 
-		elapsed := ""
-		zeroTime := time.Time{}
+	for meta.Pending > 0 {
```

Comment:
What are your thoughts on turning this into a simple `for {` loop and doing the first retrieving and outputting of checks inside the loop and put the sleep at end of the loop instead of the top? This feels like it would make this code a bit simpler.

Reply:
I'm not sure I understood completely, but I tried a refactor changing as you suggested in f1b44b4a. Is this the direction you where thinking of?